File: 38_sliding_windows/38_5_longest_good_day_streak.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int longest_good_day_streak(std::vector<int> sales) {
  int l = 0, r = 0;
  int longest_days = 0;
  while (r < sales.size()) {
    if (sales[r] >= 10) {
      if (r - l + 1 > longest_days) {
        longest_days = r - l + 1;
      }
      r++;
    } else {
      r++;
      l = r;
    }
  }

  return longest_days;
}

int main() {
  // Example1
  std::vector<int> sales1 = {0, 14, 7, 12, 10, 20};
  std::cout << longest_good_day_streak(sales1) << std::endl;

  // Example2
  std::vector<int> sales2 = {0, 14, 7, 12, 10, 20, 23, 2};
  std::cout << longest_good_day_streak(sales2) << std::endl;

  return 0;
}

File: 38_sliding_windows/38_5_longest_good_day_streak.py
def longest_good_day_streak(sales: list[int])->int:
  l, r = 0, 0
  longest_days = 0
  while r < len(sales):
    if sales[r] >= 10:
      if r - l + 1 > longest_days:
        longest_days = r - l + 1
      r += 1
    else:
      r += 1
      l = r

  return longest_days

I